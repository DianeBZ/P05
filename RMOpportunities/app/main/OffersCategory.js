import React, { Component } from 'react';
import {Text, View, StyleSheet, Dimensions, TouchableHighlight, BackAndroid} from 'react-native';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

import Offers from './Offers';

export default class OffersCategory extends Component {
  render() {
      BackAndroid.addEventListener('Back',this.onBackAndroid);
    return(
		<View>
			<View style={{height:height*0.08}}/>
			<View style = {styles.tableau}>
				<TouchableHighlight onPress={this.onPressTTCategories} style={[styles.button, { backgroundColor:'#A4D04A'}]} underlayColor="rgb(138,183,46)" >
					<Text style={styles.buttonTextTTCategorie}>Toutes catégories </Text>
				</TouchableHighlight>
				<View style={styles.button}>
					<Text style={styles.buttonText}>Produits pour la cosmétique, les arômes et parfums</Text>
				</View>
				<View style={styles.button}>
					<Text style={styles.buttonText}>Slovants</Text>
				</View>
				<View style={styles.button}>
					<Text style={styles.buttonText}>Slovants regénérés </Text>
				</View>
				<View style={styles.button}>
					<Text style={styles.buttonText}>Réactifs</Text>
				</View>
				<View style={styles.button}>
					<Text style={styles.buttonText}>Catalyseurs</Text>
				</View>
				<View style={styles.button}>
					<Text style={styles.buttonText}>Spécialités chimiques</Text>
				</View>
				<View style={styles.button}>
					<Text style={styles.buttonText}>Intermédiaires simples</Text>
				</View>
				<View style={styles.button}>
					<Text style={styles.buttonText}>Acides</Text>
				</View>
				<View style={styles.button}>
					<Text style={styles.buttonText}>Bases</Text>
				</View>
				<View style={styles.button}>
					<Text style={styles.buttonText}>Acides aminés</Text>
				</View>
			</View>
		</View>
	);
  }
  onPressTTCategories = () => {
	  BackAndroid.removeEventListener('Exit',this.onBackAndroid);
		const { navigator } = this.props;
		if (navigator) {
		  navigator.push({
			name: 'Offers',
			component: Offers,
		  })
		}
  };
  
  onBackAndroid = () => {
		const { navigator } = this.props;
		if (navigator && navigator.getCurrentRoutes().length > 1) {
			BackAndroid.removeEventListener('Back', this.onBackAndroid);
			navigator.pop();
			return true;
		} else {
			return false;
		}
  };
}



const styles = StyleSheet.create({
	tableau:{
		marginTop: width*0.1,
		alignItems : 'center',
		justifyContent:'center',
	},
	button:{
    height: width*0.11,
    width: width *0.8,
    borderRadius:5,
	borderWidth: 0.5,
	borderColor:'#A4D04A',
    justifyContent:'center',
  },
   buttonTextTTCategorie:{
      color: "white",
      textAlign: "center",
      fontSize: 20,
  },
  buttonText:{
      textAlign: "center",
      fontSize: 20,
  },
});
