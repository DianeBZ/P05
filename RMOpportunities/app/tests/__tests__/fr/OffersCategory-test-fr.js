// __tests__/OffersCategory-test-fr.js
import 'react-native';
import React from 'react';

global.lang = 'fr';
import OffersCategory from '../../../main/OffersCategory';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <OffersCategory />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
