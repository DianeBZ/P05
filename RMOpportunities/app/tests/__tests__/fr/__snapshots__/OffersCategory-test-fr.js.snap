// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F2F2F2",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 106.72,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 53.36,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "borderColor": "#A4D04A",
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "borderWidth": 0.5,
              "height": 93.38000000000001,
              "justifyContent": "center",
              "width": 675,
            },
            Object {
              "backgroundColor": "#A4D04A",
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Toutes catégories 
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#A4D04A",
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderWidth": 0.5,
          "height": 93.38000000000001,
          "justifyContent": "center",
          "width": 675,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Produits pour la cosmétique, les arômes et parfums
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#A4D04A",
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderWidth": 0.5,
          "height": 93.38000000000001,
          "justifyContent": "center",
          "width": 675,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Slovants
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#A4D04A",
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderWidth": 0.5,
          "height": 93.38000000000001,
          "justifyContent": "center",
          "width": 675,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Slovants regénérés 
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#A4D04A",
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderWidth": 0.5,
          "height": 93.38000000000001,
          "justifyContent": "center",
          "width": 675,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Réactifs
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#A4D04A",
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderWidth": 0.5,
          "height": 93.38000000000001,
          "justifyContent": "center",
          "width": 675,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Catalyseurs
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#A4D04A",
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderWidth": 0.5,
          "height": 93.38000000000001,
          "justifyContent": "center",
          "width": 675,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Spécialités chimiques
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#A4D04A",
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderWidth": 0.5,
          "height": 93.38000000000001,
          "justifyContent": "center",
          "width": 675,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Intermédiaires simples
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#A4D04A",
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderWidth": 0.5,
          "height": 93.38000000000001,
          "justifyContent": "center",
          "width": 675,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Acides
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#A4D04A",
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderWidth": 0.5,
          "height": 93.38000000000001,
          "justifyContent": "center",
          "width": 675,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Bases
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#A4D04A",
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderWidth": 0.5,
            "height": 93.38000000000001,
            "justifyContent": "center",
            "width": 675,
          },
          Object {
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderBottomWidth": 1,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Acides aminés
      </Text>
    </View>
  </View>
</View>
`;
